libraries/chain_kv/include/b1/session/bytes.hpp

```diff
@@ -0,0 +1,230 @@
+    // \remark When the counter reaches 0, the last instance will free the memory by invoking the free method on the memory allocator.
+    // \warning Assuming that this type will be accessed synchronously.
+    size_t* m_use_count_address{nullptr};
+    // The size of the memory pointed to by m_data, in bytes.
+    size_t* m_length{nullptr};
+    // A pointer to a chunk of memory.
+    void* m_data{nullptr};
+};
+
+// \brief Creates a new bytes instance with the given pointer and length.
+//
+// \tparam T The type stored in the array.
+// \tparam allocator The memory allocator type used to manage the bytes memory. This type mush implement the "memory allocator" concept.
+// \param data A pointer to an array of items to store in a bytes instance.
+// \param length The number of items in the array.
+// \param a The memory allocator used to instantiate the memory used by the bytes instance.
+template <typename T, typename allocator>
+auto make_bytes(const T* data, size_t length, allocator& a) -> bytes
+{
+    auto result = bytes{};
+    
+    length *= sizeof(T);
+    auto chunk_length = 3 * sizeof(size_t) + length;
+    auto* chunk = reinterpret_cast<char*>(a->malloc(chunk_length));
+    
+    result.m_memory_allocator_address = reinterpret_cast<size_t*>(chunk);
+    result.m_use_count_address = result.m_memory_allocator_address + sizeof(size_t);
+    result.m_length = result.m_use_count_address + sizeof(size_t);
+    result.m_data = result.m_length + sizeof(size_t);
+    
+    *(result.m_memory_allocator_address) = reinterpret_cast<size_t>(a.get());
+    *(result.m_use_count_address) = 1;
+    *(result.m_length) = length;
+    memcpy(result.m_data, reinterpret_cast<const void*>(data), length);
+    
+    return result;
+}
+
+// \brief Creates a new bytes instance with the given pointer and length.
+//
+// The returns a bytes "view" instance, in that the resulting bytes instance does not manage
+// the given memory.
+//
+// \tparam T The type stored in the array.
+// \param data A pointer to an array of items to store in a bytes instance.
+// \param length The number of items in the array.
+// \warning The resulting bytes instance does NOT take ownership of the given data pointer.
+// \warning The data pointer must remain in scope for the lifetime of the given bytes instance.
+template <typename T>
+auto make_bytes(const T* data, size_t length) -> bytes
+{
+    auto result = bytes{};
+    
+    result.m_memory_allocator_address = nullptr;
+    result.m_use_count_address = nullptr;
+    *(result.m_length) = length;
+    result.m_data = reinterpret_cast<void*>(const_cast<T*>(data));
+    
+    return result;
+}
+
+}
+
+namespace std
+{
+template <>
+class less<b1::session::bytes> final
+{
+public:
+    constexpr auto operator()(const b1::session::bytes& lhs, const b1::session::bytes& rhs) const -> bool
+    {
+        // Shorter keys are "less than" longer keys
+        if (lhs.length() < rhs.length())
```

Comment:
Other existing blob compare functions such as compare_blob() in libraries/chain/include/eosio/chain/types.hpp does lexicographic order.  ex)  "aa" comes before "z"

Reply:
I'll just try to pull that functionality in and invoke within this method call.